### DISCUSSION TITLE
Call-in segments on Azuracast - support audience interactions

### DISCUSSION THREAD

**Discussion by OrcVole** [AzuraCast/AzuraCast#7211] [Ideas]
It would be wonderful if the audience were able to phone into the show to interact with the DJ and guests and provide comment on the shows.

This could be via telephone or / and ideally through a webpage interface. Caller lines could be queued up ready for the DJ. There could be an additional role for a call-screener/producer to ensure there wasn't spam. Screening would probably require a talk-back / off-air circuit so that the radio station could interact with the members of the audience without that conversation going live on-air. 

WebRTC might be able to handle the guests through the web e.g. using Jitsi code. Phone-in might be accomplished using something like SIP / Asterisk.

**Comment by Vaalyn** [comment]
Implementing a whole SIP or audio chat stack in AzuraCast feels a bit out of scope for the project itself as well as quite a bit too complex to just bundle into AzuraCast.

I'd recommend to use existing solutions like Jitsi, MS Teams, Discord, Teamspeak, etc. to get third party people connected to a DJ so they can pull them in as needed and route the audio from those applications with something like Virtual Audio Cable to the DJs streaming software.

**Comment by Moonbase59** [comment]
Why must everything happen in a browser nowadays? I’m still quite happy with the phone-in features JACK2, IDJC and Linphone provide. Signalling, off-line talking, taking call into show, all no problem.

**Reply by OrcVole** [reply]
* https://idjc.sourceforge.io/
* https://github.com/jackaudio/jack2
* https://linphone.org/

**Reply by LordHelmchen666** [reply]
With IDJC, Jack2 setup you could also include TeamSpeak, Discord or anything like that.

**Comment by emtoonst** [comment]
You could also use a hardware solution like a Rodecaster Pro. You can connect a phone through trrs cable or bleutooth.  Works like a charm for live evenst with caller interaction.
